Makefile

```
@@ -104,11 +104,9 @@ $(K0S_GO_BUILD_CACHE):
 	mkdir -p -- '$@'
 
 .k0sbuild.docker-image.k0s: build/Dockerfile embedded-bins/Makefile.variables | $(K0S_GO_BUILD_CACHE)
-	docker build --rm \
-		--build-arg BUILDIMAGE=$(GOLANG_IMAGE) \
-		-f build/Dockerfile \
-		-t k0sbuild.docker-image.k0s build/
-	touch $@
+	docker build --iidfile '$@' \
+	  --build-arg BUILDIMAGE=$(GOLANG_IMAGE) \
+	  -t '$(@:.%=%)' - <'$<'
```

More stupid and readable variant:

```suggestion
	  -f build/Dockerfile \
	  -t k0sbuild.docker-image.k0s build/
```

Alternatively, we could use `$(patsubst .%,%,$@)` which probably is more readable?